src/main/kotlin/de/tfr/impf/ReportJob.kt

```diff
@@ -107,13 +121,89 @@ class ReportJob {
+
+                    // 2. Open modal to enter personal data
+                    if (bookingPage.isDisplayingSecondBookingStep()) {
+                        sendMessage("Second booking step is shown")
+                        Thread.sleep(2_000)
+                        bookingPage.selectEnterPersonalData()
+                        log.debug { "Open personal data modal..." }
+                    }
+
+                    // 3. Enter personal data
+                    if (bookingPage.isDisplayingPersonalDataForm()) {
+                        sendMessage("Showing personal data form")
+                        enterAndSubmitPersonalDataIntoForm(bookingPage)
+                        takeScreenshot(driver, "impf-bot-booking-data-$code.png")
+                    }
+
+                    // 4. Submit Vaccination date
+                    if (bookingPage.isDisplayingFinalBookingStep()) {
+                        sendMessage("All personal data entered, ready to submit!")
+                        log.debug { "All personal data entered, ready to submit!" }
+
+                        takeScreenshot(driver, "impf-bot-booking-overview-$code.png")
+
+                        bookingPage.submitFinalBooking()
+
+                        takeScreenshot(driver, "impf-bot-booking-confirmation-$code.png")
+                    }
+                } else {
+                    waitLongForUserInput()
+                }
             } else {
                 log.debug { "Correct code, we can't see the bookable vaccination dates: $location" }
             }
         }
     }
 
+    private fun enterAndSubmitPersonalDataIntoForm(bookingPage: BookingPage) {
+        Thread.sleep(2_000)
+        bookingPage.enterSalutation(personalDataSalutation)
+        log.info { "Salutation entered..." }
+        Thread.sleep(2_000)
+        bookingPage.enterFirstname(personalDataFirstname)
+        log.info { "Firstname entered..." }
+        Thread.sleep(2_000)
+        bookingPage.enterLastname(personalDataLastname)
+        log.info { "Lastname entered..." }
+        Thread.sleep(2_000)
+        bookingPage.enterZipCode(personalDataZipcode)
+        log.info { "Zipcode entered..." }
+        Thread.sleep(2_000)
+        bookingPage.enterCity(personalDataCity)
+        log.info { "City entered..." }
+        Thread.sleep(2_000)
+        bookingPage.enterStreet(personalDataStreet)
+        log.info { "Street entered..." }
+        Thread.sleep(2_000)
+        bookingPage.enterHousenumber(personalDataHouseNumber)
+        log.info { "Housenumber entered..." }
+        Thread.sleep(2_000)
+        bookingPage.enterPhone(personalDataMobileNumber)
+        log.info { "Phone entered..." }
+        Thread.sleep(2_000)
+        bookingPage.enterEmail(personalDataEmail)
+        log.info { "Email entered..." }
+        Thread.sleep(2_000)
+        bookingPage.submitInputWithText("Übernehmen")
+    }
+
+    private fun takeScreenshot(driver: WebDriver, fileName: String) {
```

Comment:
Should be optional, or use config, to make it compatible with the docker setup.
What happens when `Desktop` directory is missing?

Reply:
I changed that part to be optional and also made the output path changeable through a property `outputPath`. So it should be fine for Docker too.